cl/search/tasks.py

```diff
@@ -555,6 +555,45 @@ def get_doc_from_es(
     return main_doc
 
 
+def handle_ubq_retries(
+    self: Task,
+    exc: ConnectionError | ConflictError,
+    count_query=QuerySet | None,
+) -> None:
+    """Handles the retry logic for update_children_docs_by_query task based on
+    the exception received and number of documents to update.
+
+    :param self: The celery task
+    :param exc: The exception that triggered the retry.
+    :param count_query: Optional a Queryset to retrieve the number of docs to
+    update.
+    :return: None
+    """
+
+    retry_count = self.request.retries
+    if retry_count >= self.max_retries:
+        raise exc
+
+    if isinstance(exc, ConnectionError) and count_query:
+        num_documents = count_query.count()
+        estimated_time_ms = num_documents * 15  # 15ms per document
+        # Convert ms to seconds
+        estimated_delay = round(estimated_time_ms / 1000)
+        # Apply exponential backoff adding a small random component.
+        min_delay = max(estimated_delay, 10)
+        rand_component = randint(10, 30)
+        countdown = ((retry_count + 1) * min_delay) + rand_component
+    else:
+        # Default case for ConflictError
+        min_delay = 10  # 10 seconds
```

Comment:
I'd keep units in the variable names. e.g. call this `min_delay_sec` and change the others above and below.

Reply:
suggestion applied, thanks!